data/ui_scripts/server_list/serverlist.lua

```diff
@@ -182,20 +551,76 @@ SystemLinkJoinMenu.MakeText = function(menu, f5_arg1, text, color)
 end
 
 function menu_systemlink_join(f19_arg0, f19_arg1)
-	local width = 1145
+	if not file_exists("players2/favourites.txt") then
+		local file = io.open("players2/favourites.txt", "w")
+		file:close()
+	end
+
+	if not file_exists("players2/blacklisted.txt") then
+		local file = io.open("players2/blacklisted.txt", "w")
+		file:close()
+	end
 
+	Engine.SetDvarFromString("ui_sorting", "2")
+	Engine.SetDvarFromString("ui_server_type", "public")
```

Comment:
Same as above, shouldn't be done in lua, also I'm not sure why you're using dvars

Reply:
Did not find a better way to have shared variables.